JSON serialization of InputMedia Types

joshuastorch opened the pull request:
The InputMedia Classes were not serialized as JSON and instead printed as default data class strings. This implements a possible fix for that problem.
This Problem was reported in issue #213

vjgarciag96 reviewed (approved):
Thanks for the fix @joshuastorch! (and sorry for the late review)

LGTM. We usually create retrofit converters like https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/main/telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt rather than overriding the `toString` for serialisation purposes. But we can go ahead with the fix and move the change to a retrofit converter as a separated cleanup task.

joshuastorch commented:
I resolved the merge conflict.
I have little knowlege about retrofit and based the toString overrides on the ReplyMarkup Types like [InlineKeyboardMarkup](https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/main/telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/InlineKeyboardMarkup.kt). 

I looked into the retrofit converters you mentioned but as far as i can see they are not called when using functions like [editMessageMedia](https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/main/telegram/src/main/kotlin/com/github/kotlintelegrambot/network/ApiClient.kt#L1062). If they were i think the [GsonConverter](https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/main/telegram/src/main/kotlin/com/github/kotlintelegrambot/network/ApiClient.kt#L135) would take care of it.

joshuastorch commented:
I added a new ```InputMediaConverterFactory``` and removed the ```toString``` methods. 
I had to add the ```TelegramFileAdapter``` and ```InputMediaAdapter``` to ```GsonFactory.createForApiClient()```

vjgarciag96 commented on telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/inputmedia/InputMedia.kt:
```diff
@@ -76,8 +76,8 @@ data class InputMediaAnimation(
 ) : InputMedia() {
-    override val type: String
-        get() = InputMediaTypes.ANIMATION
+    @SerializedName(InputMediaFields.TYPE)
```
Nice one! I imagine we can now remove the code handling this manually in `InputMediaAdapter`?https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/main/telegram/src/main/kotlin/com/github/kotlintelegrambot/network/serialization/adapter/InputMediaAdapter.kt#L24-L25

joshuastorch replied:
I think the entire custom serializer ```InputMediaAdapter``` for the ```InputMedia``` types can be removed.

vjgarciag96 reviewed (approved):
This looks great! Thank you so much for the contribution! Let's merge it
